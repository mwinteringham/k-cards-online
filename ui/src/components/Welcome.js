import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Form from 'react-bootstrap/Form'
import Button from 'react-bootstrap/Button';

function Welcome() {

    return (
        <div>
            <Row className="mb-5">
            <Col>
                <h1>Welcome to K-Cards online</h1>
            </Col>
            </Row>
            <Row>
            <Col />
            <Col>
                <h2>Join a Workshop</h2>
                <p>Enter your name and the workshop code to Join Workshop</p>
                <Form>
                <Form.Control className="mb-3" placeholder="Your name" />
                <Form.Control className="mb-3" placeholder="Workshop code" />
                <Button type="submit">Join Workshop</Button>
                </Form>
            </Col>
            <Col>
                <h2>Host a Workshop</h2>
                <p>Enter the name of the workshop and click Start Workshop</p>
                <Form>
                <Form.Control placeholder="Workshop name" className="mb-3" />
                <Button className="mt-5">Start Workshop</Button>
                </Form>
            </Col>
            <Col />
            </Row>
            <Row>
                <Col>
                
                </Col>
            </Row>
        </div>
    )

}

export default Welcome;