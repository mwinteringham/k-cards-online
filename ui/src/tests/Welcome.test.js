import { render, fireEvent, screen, waitFor} from '@testing-library/react';
import Welcome from '../components/Welcome';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import axios from 'axios';

var MockAdapter = require('axios-mock-adapter');

const mock = new MockAdapter(axios);
const history = createMemoryHistory();

test('renders the Welcome component', () => {
  const welcomeComponent = render(<Welcome />);
  
  expect(welcomeComponent.baseElement).toMatchSnapshot();
});

test('creating a new workshop takes you to a new page', async () => {
  mock.onPost('/workshop', { name: 'BREWT' }).reply(201, { code : 'abc'});

  render(<Router history={history}><Welcome /></Router>)

  fireEvent.change(screen.getByTestId(/workshopName/i), {
    target: {value: 'BREWT'}
  })
  fireEvent.submit(screen.getByTestId(/startWorkshop/i));

  await waitFor(() => expect(history.location.pathname).toBe('/host'));
});

test('joining a new workshop takes you to a new page', async () => {
  mock.onPost('/workshop/abc/join', { name: 'Mark' }).reply(201, { code : 'abc'});

  render(<Router history={history}><Welcome /></Router>)

  fireEvent.change(screen.getByTestId('workshopCode'), {
    target: {value: 'abc'}
  });
  fireEvent.change(screen.getByTestId(/attendeeName/i), {
    target: {value: 'Mark'}
  });
  fireEvent.submit(screen.getByTestId(/joinWorkshop/i));

  await waitFor(() => expect(history.location.pathname).toBe('/attendee'));
});

test('submitting a blank workshop name returns an error', async () => {
  render(<Welcome />);

  fireEvent.submit(screen.getByTestId(/startWorkshop/i));

  expect(screen.getByTestId('hostError')).toHaveTextContent('Please enter a name for your workshop');
});

test('submitting a blank workshop connection returns an error', async () => {
  render(<Welcome />);

  fireEvent.submit(screen.getByTestId(/joinWorkshop/i));
  expect(screen.getByTestId('usernameError')).toHaveTextContent('Please enter a name that has at least two characters');
  expect(screen.getByTestId('workshopCodeError')).toHaveTextContent('Please enter a valid workshop code');

});

test('submitting an invalid workshop code returns an error', async () => {
  mock.onPost('/workshop/def/join', { name: 'Geoff' }).reply(404);

  render(<Welcome />);

  fireEvent.change(screen.getByTestId('workshopCode'), {
    target: {value: 'def'}
  });
  fireEvent.change(screen.getByTestId(/attendeeName/i), {
    target: {value: 'Geoff'}
  });
  fireEvent.submit(screen.getByTestId(/joinWorkshop/i));

  await waitFor(() => expect(screen.getByTestId('workshopCodeError')).toHaveTextContent('Please enter a valid workshop code'));
});
