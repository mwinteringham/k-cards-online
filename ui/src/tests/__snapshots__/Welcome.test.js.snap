// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Welcome component 1`] = `
<body>
  <div>
    <div>
      <div
        class="text-center mt-5 container-fluid"
      >
        <div
          class="mb-5 row"
        >
          <div
            class="col"
          >
            <h1>
              Welcome to K-Cards online
            </h1>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          />
          <div
            class="col"
          >
            <h2>
              Join a Workshop
            </h2>
            <p>
              Enter your name and the workshop code to Join Workshop
            </p>
            <form
              class=""
            >
              <input
                class="mb-3 form-control"
                data-testid="attendeeName"
                placeholder="Your name"
              />
              <input
                class="mb-3 form-control"
                data-testid="workshopCode"
                placeholder="Workshop code"
              />
              <button
                class="btn btn-primary"
                data-testid="joinWorkshop"
                type="button"
              >
                Join Workshop
              </button>
            </form>
          </div>
          <div
            class="col"
          >
            <h2>
              Host a Workshop
            </h2>
            <p>
              Enter the name of the workshop and click Start Workshop
            </p>
            <form
              class=""
            >
              <input
                class="mb-3 form-control"
                data-testid="workshopName"
                placeholder="Workshop name"
              />
              <button
                class="mt-5 btn btn-primary"
                data-testid="startWorkshop"
                type="button"
              >
                Start Workshop
              </button>
            </form>
          </div>
          <div
            class="col"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
